system/Helpers/form_helper.php

```diff
@@ -0,0 +1,859 @@
+<?php
+
+// -------------------------------------------------------------------------------
+
+if ( ! function_exists('form_open'))
+{
+    /**
+     * Form Declaration
+     *
+     * Creates the opening portion of the form.
+     *
+     * @param	string	the URI segments of the form destination
+     * @param	array	a key/value pair of attributes
+     * @param	array	a key/value pair hidden data
+     * @return	string
+     */
+    function form_open(string $action = '', array $attributes = [], array $hidden = []): string
+    {
+        // If no action is provided then set to the current url
+        if ( ! $action)
+        {
+            $action = current_url(true);
+        }
+        // If an action is not a full URL then turn it into one
+        elseif (strpos($action, '://') === false)
+        {
+            $action = site_url($action);
+        }
+        $attributes = _attributes_to_string($attributes);
+        if (stripos($attributes, 'method=') === false)
+        {
+            $attributes .= ' method="post"';
+        }
+        if (stripos($attributes, 'accept-charset=') === false)
+        {
+            $attributes .= ' accept-charset="'.strtolower(config_item('charset')).'"';
+        }
+        $form = '<form action="'.$action.'"'.$attributes.">\n";
+        
+        // Add CSRF field if enabled, but leave it out for GET requests and requests to external websites
+        $getCsrf = new \Config\Filters();
+        $before = $getCsrf->globals->before;
+        if ((in_array('csrf', $before) || array_key_exists('csrf', $before)) &&
+                strpos($action, base_url()) !== false && ! stripos($form, 'method="get"'))
+        {
+            $security = \Config\Services::security();
+            $hidden[$security->getCsrfTokenName()] = $security->getCsrfHash();
```

Comment:
Methods names need to have CSRF capitalized: `getCSRFTokenName()` and `getCSRFHash()`.